Change u₀ to u0

TorkelE opened the pull request:
Related to https://github.com/SciML/ReactionNetworkImporters.jl/issues/104

This is highly breaking though.

isaacsas commented:
If we want to add this there needs to be an overload so the old notation works and returns `u0` too. Then this isn’t breaking anything.

TorkelE commented:
Is that possible (without simultaneously supporting both fields)? I am only familiar with overloads for functions, so would be unsure how to do this.

isaacsas commented:
See `getproperty` and `setproperty`.

TorkelE commented:
But both `u₀` and `u0` which are given to these would have the same type (Symbol). Or you mean create new version of these functions on `ParsedReactionNetwork` that first check if the input is `u₀`, and if so calls it again with input `u0`?

TorkelE commented:
Updated. These overloads added:
```julia
# Ensures that `prnbng.u₀` works.
function Base.getproperty(prnbng::ParsedReactionNetwork, name::Symbol)
    if name === :u₀
        return getfield(prnbng, :u0)
    else
        return getfield(prnbng, name)
    end
end

# Ensures that `prnbng.u₀ = ...` works.
function Base.setproperty!(prnbng::ParsedReactionNetwork, name::Symbol, x)
    if name === :u₀
        return setfield!(prnbng, :u0, x)
    else
        return setfield!(prnbng, name, x)
    end
end
```
Given that parsed reaction networks are immutable, I don;t think the other one is needed though? 

I have also added a couple of tests in the tests file to check that (for some loaded network)
```julia
@test isequal(prnbng.u0, prnbng.u₀)
```

isaacsas commented on src/u0_index_change.jl:
```diff
@@ -0,0 +1,17 @@
+# Ensures that `prnbng.u₀` works.
+function Base.getproperty(prnbng::ParsedReactionNetwork, name::Symbol)
+    if name === :u₀
+        return getfield(prnbng, :u0)
+    else
+        return getfield(prnbng, name)
+    end
+end
+
+# Ensures that `prnbng.u₀ = ...` works.
+function Base.setproperty!(prnbng::ParsedReactionNetwork, name::Symbol, x)
+    if name === :u₀
+        return setfield!(prnbng, :u0, x)
+    else
+        return setfield!(prnbng, name, x)
+    end
+end
```
I think these can just be in the ReactionNetworkImporters.jl file. No need for a separate file.

isaacsas reviewed (changes requested):
One change then I'm happy to merge. Thanks!

TorkelE commented:
This one should be ready now.
